doc_test.go

```diff
@@ -65,6 +66,41 @@ func TestName(t *testing.T) {
 	}
 }
 
+func TestOpenByName(t *testing.T) {
```

Comment:
Would you mind adding some comments on what is the idea here? What kind of use-case would need to do something like this? Seems like it use the lib only to get a reference onthe tty file, which is already opened, then re-opened? As I don't really understand what is going on, it would be good to have some comments to provide context.

Reply:
I ran into this issue in the [delve tests](https://github.com/go-delve/delve/blob/master/service/debugger/debugger_unix_test.go#L57), where the test code opens a pty, then passes `tty.Name()` through to the code under test which opens a TTY from the given filename. I've added a comment.